src/Proxy.hs

```diff
@@ -1227,21 +1283,23 @@ getAccountTransactionsWorker :: IncludeMemos -> Text -> Handler TypedContent
 getAccountTransactionsWorker includeMemos addrText = do
   i <- internationalize
   -- Get the canonical address of the account.
-  -- This fails only if we cannot get an understandable response from the GRPC, i.e., if the node is not reachable,
-  -- or if it returns invalid JSON.
-  let getAddress k =
-        case addressFromText addrText of
-          Left _ -> respond400Error EMMalformedAddress RequestInvalid
-          Right givenAddr -> do
-            let doGetAccAddress = do
-                  lastFinBlock <- getLastFinalBlockHash
-                  ai <- getAccountInfo addrText lastFinBlock
-                  case ai of
-                    Right (GRPCResponse hds Null) -> return $ Right $ GRPCResponse hds givenAddr -- the account does not exist on the node, assume the given address is the one we want.
-                    Right (GRPCResponse hds val) -> return $ Right $ GRPCResponse hds (fromRight givenAddr (parseEither (withObject "account info" (.: "accountAddress")) val))
-                    -- This should not happen, accountInfo always returns valid JSON
-                    Left err -> return $ Left err
-            runGRPC doGetAccAddress k
+  let getAddress k = do
+        givenAddr <- doParseAccountAddress (Just "getAccountTransactionsWorker") addrText
+        let doGetAccAddress = do
+              ai <- getAccountInfo (AccAddress givenAddr) LastFinal
+              return $ case ai of
+                StatusOk (GRPCResponse hds (Right accInfo)) ->
+                  StatusOk $ GRPCResponse hds $ Right (aiAccountAddress accInfo)
+                StatusOk (GRPCResponse hds (Left err)) ->
+                  StatusOk $ GRPCResponse hds $ Left err
+                -- FIXME: Headers should be propagated even on a non-OK status.
```

Comment:
Don't leave FIXME's in the code. THere is already an issue in the client for this. If that is addressed then changes will have to be propagated.

Reply:
Ok 👍🏼